src/interactive/scala/tools/nsc/interactive/InteractiveReporter.scala

```diff
@@ -18,7 +18,7 @@ abstract class InteractiveReporter extends Reporter {
   val otherProblems = new ArrayBuffer[Problem]
 
   override def info0(pos: Position, msg: String, severity: Severity, force: Boolean): Unit = try {
-    severity.count += 1
+    severity.count.incrementAndGet()
```

Comment:
for repl, interactive, runtime reflection, I think that they can be single threaded if needed. Lower priority at least

Thoughts?